### DISCUSSION TITLE
[Question] How to build koreader with DEBUG switch off?

### DISCUSSION THREAD

**Discussion by zhaokoh** [koreader/koreader#12092] [Q&A] [answered]
Hi koreader devs,

I am really impressed with your work and please keep it up. I am trying to build a koreader instance for my Mac with no DEBUG in the console, I tried:

```
./kodev build --no-debug
```

but when I run the kodev using the command ```./kodev run```, there are still DEBUG messages in the console.

I don't think the debug switch took effect - perhaps I used it incorrectly?

p.s.: Also, the progress sync server seems to giving this message whenever I try to register a new user (therefore failed):

POST https://sync.koreader.rocks:443/users/create
wantread
06/26/24-12:48:38 DEBUG KOSyncClient:register failure: "wantread not expected" 

Thanks

**Comment by benoit-pierre** [comment, marked as answer]
`--no-debug` is for building a release build: stripped native libraries, no asserts, release optimized flags. It has no impact on debug traces (`kodev run` always start koreader with `-d`).

Additionally, you should use `./kodev run -n` too (the short variant of `--no-debug` which does not work for `run` because of a bug), or you may end up compiling and running a debug build (the default mode for the emulator target).

For the sync server error, I think it's the same issue as for the nightlies: see discussion in https://github.com/koreader/koreader/issues/12073.

**Reply by benoit-pierre** [reply]
And if you want to build an application bundle for macOS, use `./kodev release macos`.

**Reply by zhaokoh** [reply]
Wow thank you so much @benoit-pierre for your quick response! 🙏

**Comment by zhaokoh** [comment]
@benoit-pierre I tried ```./kodev release macos``` but get the following error.

```
(base) TBook:koreader Zhao$ ./kodev release -d macos
************ CMAKE_MAKE_PROGRAM: /Library/Developer/CommandLineTools/usr/bin/make **********
************ NINJA:  **********
************ MAKE: /Library/Developer/CommandLineTools/usr/bin/make -j10 -l10 **********
git submodule update --remote l10n
Unsupported target for release: divh.

usage: release <OPTIONS> <TARGET>
```

**Reply by benoit-pierre** [reply]
Did you follow the instructions in `doc/Building.md` for setting up a build environment?

**Reply by benoit-pierre** [reply]
Note: if you just want a build of the current master for macOS, you can get one from Github Actions: https://github.com/koreader/koreader/actions/runs/9668550154.

**Reply by zhaokoh** [reply]
Hmm, downloaded the current master build for macOS, whenever I tried to open a PDF file, it shows the following.

<img width="603" alt="image" src="https://github.com/koreader/koreader/assets/10231698/7fa72fe9-e9ac-48d2-8e6b-076451315182">

My local build has no issue opening a PDF file. 

For releasing a local build, I need XCode installed, but I am now fighting with the dynamic libraries - for libsodium, I tried to symlink from ```libs/libsodium.23.dylib``` to ```libs/libsodium.26.dylib```. But the following errors I have no idea how to fix it.

```
~/Development/external/koreader/koreader-macos-arm64-apple-darwin23.2.0/bundle/Contents/koreader ~/Development/external/koreader
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/install_name_tool: warning: changes being made to the file will invalidate the code signature in: libs/libintl.8.dylib
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/install_name_tool: warning: changes being made to the file will invalidate the code signature in: libs/libsodium.23.dylib
error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/install_name_tool: can't open file: libs/libglib-2.0.dylib (No such file or directory)
error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/install_name_tool: can't open file: libs/libfmq.1.dylib (No such file or directory)
error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/install_name_tool: can't open file: libs/libzyre.1.dylib (No such file or directory)
```

So close and yet so far 😂

**Reply by benoit-pierre** [reply]
> Hmm, downloaded the current master build for macOS, whenever I tried to open a PDF file, it shows the following.

Can you get more info? Some traces?

Also, maybe try an older build: https://github.com/koreader/koreader/actions?query=branch%3Amaster (ignore `master` branches from fork, like `notmarek:master`).


`libsodium` should not be needed anymore. You're building the current master? Submodules are up-to-date?

**Reply by benoit-pierre** [reply]
The MuPDF issue has been resolved, current master should work.
